src/prim_main.F90

```diff
@@ -347,15 +347,14 @@ end subroutine finalize_hommexx_session
                                          elem_state_Qdp_ptr, elem_state_Q_ptr, elem_state_ps_v_ptr,                    &
                                          elem_derived_omega_p_ptr)
           endif
-          
+#else
+          call prim_run_subcycle(elem, hybrid,nets,nete, tstep, tl, hvcoord,1)
+#endif
           if(.not. disable_diagnostics) then
-            if (MODULO(tl%nstep,statefreq)==0) then
+            if (MODULO(tl%nstep,statefreq)==0 .or. tl%nstep >= nstep) then
```

Comment:
My suggestion was wrong in an important detail. We don't want `>= nstep` here; rather, we want `>= nEndStep`. The first will run `printstate` at each I/O point, which is wrong; the second will be sure to to run it at least at the final step, which is what we want. Sorry for the error.

This fix needs to be propagated to the `subcycle` routine's logic.

Reply:
I can fix that.